core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java

```diff
@@ -61,11 +59,16 @@ public void checkPermission(AuthorizationPermission perm, String role) {
    }
 
    public void checkPermission(AuthorizationConfiguration configuration, AuthorizationPermission perm) {
-      checkPermission(configuration, perm, null);
+      checkPermission(configuration, perm, (String)null);
    }
 
    public void checkPermission(AuthorizationConfiguration configuration, AuthorizationPermission perm,
-         String role) {
+                               String role) {
+      checkPermission(configuration, perm, Collections.singletonList(role));
```

Comment:
Unfortunately this constructs a new collection for every invocation even when we don't need to check for a role (which is the majority of cases in the code). In the above invocations it would be better to substitute (String)null with Collections.emptyList().